platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/GeoJsonSource.java

```diff
@@ -247,22 +277,46 @@ public void setGeoJson(String json) {
    * Updates the url
    *
    * @param url the GeoJSON FeatureCollection url
+   * @deprecated use {@link #setUri(URI)} instead
    */
+  @Deprecated
   public void setUrl(@NonNull URL url) {
     checkThread();
     setUrl(url.toExternalForm());
   }
 
+  /**
+   * Updates the uri
+   *
+   * @param url the GeoJSON FeatureCollection uri
+   */
+  public void setUri(@NonNull URI url) {
```

Comment:
this javadoc could benefit of some examples: 
 - `http://`
 -  `asset://`
 - `file://`